core/container/src/main/java/org/wildfly/swarm/internal/FileSystemLayout.java

```
@@ -76,6 +79,40 @@ public static FileSystemLayout create() {
+        String implClassName = System.getProperty(FileSystemLayout.class.getName());
+        if (implClassName != null) {
+            implClassName = implClassName.trim();
+            if (!implClassName.isEmpty()) {
+                FileSystemLayout layout = null;
+                // Attempt to load the specified class implementation.
+                ClassLoader loader = Thread.currentThread().getContextClassLoader();
+                try {
+                    Class<?> clazz = loader.loadClass(implClassName);
+                    // Check if clazz is an implementation of FileSystemLayout or not.
+                    if (FileSystemLayout.class.isAssignableFrom(clazz)) {
+                        Class<? extends FileSystemLayout> implClazz = clazz.asSubclass(FileSystemLayout.class);
+                        // Check if we have an appropriate constructor or not.
+                        Constructor<? extends FileSystemLayout> ctor = implClazz.getDeclaredConstructor(String.class);
+                        layout = ctor.newInstance(root);
+                    } else {
+                        String msg = String.format("%s does not subclass %s", implClassName, FileSystemLayout.class.getName());
+                        LOG.warn(SwarmMessages.MESSAGES.invalidFileSystemLayoutProvided(msg));
+                    }
+                } catch (ReflectiveOperationException e) {
+                    String msg = String.format("Unable to instantiate layout class (%s) due to %s", implClassName, e.getMessage());
+                    LOG.warn(SwarmMessages.MESSAGES.invalidFileSystemLayoutProvided(msg));
+                    LOG.debug(SwarmMessages.MESSAGES.invalidFileSystemLayoutProvided(msg), e);
+                }
+
+                if (layout != null) {
+                    return layout;
+                }
+            } else {
+                LOG.warn(SwarmMessages.MESSAGES.invalidFileSystemLayoutProvided("Implementation class name is empty."));
```

At this point we know user wants to leverage the property; wouldn't it make more sense to blow up with exception rather then having a warning (e.g. fail fast)?

---

Good point. I did not want to break the existing processes. Should we rather bail when the exception is raised? I log a warning and continue.

---

Well, you have the use-case for this scenario. Does it make sense for you to continue operating if the properly is malformed?
Either way, as long as the behaviour is documented, it should be fine.

---

Done.

1. Making use of a constant called:`thorntail.filesystem.layout.class`
2. I raise an exception if the class initialization fails.

---

I know @manovotn asked for this, so this is not really asking for change, just wondering -- should we start introducing the word `thorntail` into thins like configuration keys etc.? So far, everything has been `swarm`, which is legacy, but consistent.

---

I wasn't sure about the naming convention which is why I had decided to go with the classname instead 😉. I can switch to `swarm....` if you prefer.

My preference would be to cut a release where *all* previous keys were *also* referenced as new keys. That way, we do not know have to worry about which keys are considered as old and which are considered as new. But then again, that would also mean that there was a convenient utility in place for getting the preference value from different data sources (YML, SysProps, SysEnv, etc.).